### DISCUSSION TITLE
Wording of "Close Terminal?" prompt

### DISCUSSION THREAD

**Discussion by peterjc** [ghostty-org/ghostty#3489] [Feature Requests, Ideas]
1. Using macOS
2. Run Ghostty v1.0
3. ...
4. Open new tab with Apple+T
5. Click the "X" in the left of the tab controls OR press Apple+W
6. Get the following popup:

> Close Terminal?
> The terminal still has a running process.
> If you close the terminal the process will be killed.
> [Close the Terminal]
> [Cancel]

Or,

1. Using macOS
2. Run Ghostty v1.0
3. ...
4. Open new split right with Apple+D
5. Press Apple+W
6. Get the same popup.

The fact I get prompted at all is a separate issue, see https://github.com/ghostty-org/ghostty/discussions/3070 - assume for the sake of argument that I have something running in the terminal.

My issue is with the wording, and that "Terminal" can be open to interpretation, especially as a new user.

I think it could be clearer that this means "Tab" or "Terminal Tab", or if using panes, "Pane" or "Terminal Pane".

Initially I was unsure if this might mean "Terminal Window" (or worse, "Terminal Application" - meaning all the Ghostty windows/tabs/panes).

**Comment by peterjc** [comment]
Very closely related to this (and I was torn on a comment here versus a new issue), the main menu under "File" has:

* New Window (Apple+N)
* New Tab (Apple+T)
* --
* Split right (Apple+D)
* Split down (Shift+Apple+D)
* --
* Close (Apple+W)
* Close Windows (Shift+Apple+W)
* Close All Windows (Alt+Shift+Apple+W)

[And yes, you can write the shortcuts as Command+W etc - hopefully this still clear.]

Might "Close (Apple+W)" say "Close Tab (Apple+W)" (which would match browsers like Firefox) or "Close Pane (Apple+W)" as appropriate?

**Comment by hlascelles** [comment]
As a comparison, Terminator just has "Close" in the Right Click Menu which could be better.

Agree, it would be great to have "Close Window, Close Tab, Close Pane" all as explicit things. I suppose "Close All Windows" too.

**Reply by pluiedev** [reply]
Close window, close pane (close split) and close tab are already implemented in GTK on tip, although I'm not sure if that's the case in macOS
